packages/amplify-cli/src/extensions/amplify-helpers/auth-notifications.ts

```diff
@@ -0,0 +1,92 @@
+import { $TSAny, $TSContext, stateManager, exitOnNextTick, FeatureFlags, pathManager } from 'amplify-cli-core';
+import { collectDirectivesByType, readProjectConfiguration } from 'graphql-transformer-core';
+import { printer, prompter } from 'amplify-prompts';
+import { DocumentNode, FieldNode, DirectiveNode, parse } from 'graphql';
+import path from 'path';
+import fs from 'fs-extra';
+
+async function flipNotificationFlag(projectPath: string, flagName: string, value: boolean): Promise<void> {
+  let config = stateManager.getCLIJSON(projectPath, undefined, {
+    throwIfNotExist: false,
+    preserveComments: true,
+  });
+
+  config.features.graphqltransformer[flagName] = value;
+  stateManager.setCLIJSON(projectPath, config);
+  await FeatureFlags.reloadValues();
+}
+
+export async function notifyFieldAuthSecurityChange(context: $TSContext): Promise<void> {
+  const flagName = 'showfieldauthnotification';
+  const dontShowNotification = !FeatureFlags.getBoolean(`graphqltransformer.${flagName}`);
+
+  if (dontShowNotification) return;
+
+  const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+  const meta = stateManager.getMeta();
+
+  const apiNames = Object.entries(meta?.api || {})
+    .filter(([_, apiResource]) => (apiResource as $TSAny).service === 'AppSync')
+    .map(([name]) => name);
+
+  const doesNotHaveGqlApi = apiNames.length < 1;
+
+  if (doesNotHaveGqlApi) {
+    return await flipNotificationFlag(projectPath, flagName, false);
+  }
+
+  const apiName = apiNames[0];
+  const apiResourceDir = pathManager.getResourceDirectoryPath(projectPath, 'api', apiName);
+  const project = await readProjectConfiguration(apiResourceDir);
+  const directiveMap = collectDirectivesByType(project.schema);
+  const doc: DocumentNode = parse(project.schema);
+  const fieldDirectives: Set<string> = haveFieldAuthDirectives(doc);
+
+  if (displayAuthNotification(directiveMap, fieldDirectives)) {
+    printer.blankLine();
+    const continueChange = await prompter.yesOrNo(
+      `This version of Amplify CLI introduces additional security enhancements for your GraphQL API. ` +
+        `The changes are applied automatically with this deployment. This change won't impact your client code. Continue`,
+    );
+
+    if (!continueChange) {
+      await context.usageData.emitSuccess();
+      exitOnNextTick(0);
+    }
+
+    const schemaPath = path.join(apiResourceDir, 'schema.graphql');
+    fs.appendFile(schemaPath, ' ');
+  }
+
+  await flipNotificationFlag(projectPath, flagName, false);
+}
+
+export function displayAuthNotification(directiveMap: any, fieldDirectives: Set<string>): boolean {
+  return Object.keys(directiveMap).some((typeName: string) => {
+    const typeObj = directiveMap[typeName];
+    const modelDirective = typeObj.find((dir: DirectiveNode) => dir.name.value === 'model');
+
+    const subscriptionOff = modelDirective.arguments.some((arg: any) => {
```

Comment:
it's possible there might be a schema with no `@model`s
```suggestion
    const subscriptionOff = modelDirective?.arguments?.some((arg: any) => {
```